core/src/main/java/org/infinispan/factories/SizeCalculatorFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.infinispan.factories;
+
+import org.infinispan.configuration.cache.MemoryConfiguration;
+import org.infinispan.configuration.cache.MemoryConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.infinispan.container.KeyValueMetadataSizeCalculator;
+import org.infinispan.container.entries.CacheEntrySizeCalculator;
+import org.infinispan.container.entries.PrimitiveEntrySizeCalculator;
+import org.infinispan.container.offheap.OffHeapEntryFactory;
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.marshall.core.WrappedByteArraySizeCalculator;
+
+/**
+ * Factory for creating size calculator used to estimate size of objects
+ * @author wburns
+ * @since 9.0
+ */
+@DefaultFactoryFor(classes = KeyValueMetadataSizeCalculator.class)
+public class SizeCalculatorFactory extends AbstractNamedCacheComponentFactory implements AutoInstantiableFactory {
+   @Override
+   public <T> T construct(Class<T> componentType) {
+      MemoryConfiguration memory = configuration.memory();
+      if (memory.evictionType().isMemoryBased()) {
+         StorageType type = memory.storageType();
+         switch (type) {
+            case BINARY:
+               return componentType.cast(CacheEntrySingleton.INSTANCE);
+            case OFF_HEAP:
+               return componentType.cast(componentRegistry.getComponent(OffHeapEntryFactory.class));
+            case OBJECT:
```

Comment:
The case can be removed. It will always fall into `default`. Although *with* a case it's a bit more readable...

Reply:
Yeah I am aware, I more put it there so the comment explained why it fell into default is all.